src/pages/Home/Home.page.jsx

```diff
@@ -1,28 +1,63 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { ThemeProvider } from 'styled-components';
 import Navbar from '../../components/Navbar';
 import { GlobalStyles } from '../../components/globalStyles';
 import { lightTheme, darkTheme } from '../../components/Themes';
 import Card from '../../components/Card';
-import { youtube } from '../../utils/youtube-videos-mock';
 import { ContainerList } from './Home.styles';
+// import { youtube } from '../../utils/youtube-videos-mock';
+import { getVideos } from '../../services/getVideos';
+import VideoDetail from '../VideoDetails';
 
 function Home() {
   const [darkMode, setDarkMode] = useState(true);
-  const cardList = youtube.items.map((video) => (
-    <Card
-      key={video.etag}
-      image={video.snippet.thumbnails.high.url}
-      title={video.snippet.title}
-      description={video.snippet.description}
-    />
-  ));
+  const [param, setParam] = useState('wizeline');
+  const [listofVideos, setListofVideos] = useState([]);
+  const [error, setError] = useState();
+
+  const [selectedVideo, setSelectedVideo] = useState({});
+  const [seeVideoDetail, setSeeVideoDetail] = useState(false);
+
+  useEffect(() => {
+    getVideos(param, setListofVideos, setError);
+    // setListofVideos(youtube.items);
+    // setError(undefined);
+  }, [param]);
+
+  const showContent = () => {
+    if (error) {
+      return <p id="errorMsg">Error getting the data from the server</p>;
+    }
+    if (listofVideos.length === 0) {
+      return <p id="errorMsg">No videos to show</p>;
+    }
+    return listofVideos
+      .map((video) => (
+        <Card
+          key={video.etag}
+          image={video.snippet.thumbnails.high.url}
+          title={video.snippet.title}
+          description={video.snippet.description}
+          video={video}
+          setSelectedVideo={setSelectedVideo}
+          setSeeVideoDetail={setSeeVideoDetail}
+        />
+      ))
+      .splice(1);
+  };
```

Comment:
This function in reality is a component, I would create the component in is own file
In the next lines you have something like:
```jsx
{!seeVideoDetail && <ContainerList>{showContent()}</ContainerList>}
{seeVideoDetail && <VideoDetail {...props} />
```
In the first case, when `seeVideoDetail` is false then you show the `Default` or the `Home` component, I would move the code and create something like:

```jsx
function DefaultComponent({ error, listofVideos }) {
  if (error) {
    return <p id="errorMsg">Error getting the data from the server</p>;
  }
  if (listofVideos.length === 0) {
    return <p id="errorMsg">No videos to show</p>;
  }

  return (
    <ContainerList>
      {listofVideos
        .splice(1)
        .map((video) => (
          <Card
            key={video.etag}
            image={video.snippet.thumbnails.high.url}
            title={video.snippet.title}
            description={video.snippet.description}
            video={video}
            setSelectedVideo={setSelectedVideo}
            setSeeVideoDetail={setSeeVideoDetail}
          />
        ))}
    </ContainerList>
  );
}
```
and with this component:

```jsx
{!seeVideoDetail && <DefaultComponent {...props} />}
{seeVideoDetail && <VideoDetail {...props} />
```

I think this is more expressive and you know what's going on at first sight, moving little bits of code that makes sense on its own improves readability and maintainability, it's just my opinion 😬

I now this will change with the router, but I hope this is useful for you in some way, good work 👍